### DISCUSSION TITLE
Add controlled component capabilities to the Popover component

### DISCUSSION THREAD

**Discussion by JoseRFelix** [tailwindlabs/headlessui#2206] [Ideas]
## Motivation

This feature will enhance the library's functionality and cater to more use cases.

## Description:

Currently, the Popover component in the @headlessui/react library does not have the capability to be a controlled component. This means that it cannot be opened or closed programmatically by updating the parent component's state.

It would be great if this capability could be added to the Popover component for more advanced use cases.

## Proposed Solution:

- Add a new prop called `isOpen` to the Popover component. This prop will accept a boolean value and determine whether the popover is open or closed.
- Add a new method called `onChange()` that receives the next state value.
- Pass `togglePopover()` to the render function

**Comment by MiroslavPetrik** [comment]
I believe this is duplicate of https://github.com/tailwindlabs/headlessui/discussions/1626

**Reply by gregg-cbs** [reply]
Duplicate of https://github.com/tailwindlabs/headlessui/discussions/2360 too

**Comment by gregg-cbs** [comment]
This would make the Popover a whole lot more versatile and easier to use in these situations. Any response from the headlessui team?
